src/app/core/ui/navigation/navigation/navigation.component.ts

```diff
@@ -104,11 +107,29 @@ export class NavigationComponent {
       this.CONFIG_ID,
     );
 
-    this.menuItems = await this.routePermissionService.filterPermittedRoutes(
-      config.items,
-    );
+    const menuItems = config.items.map((item) => {
+      if ("entityType" in item) {
+        return this.getMenuItem(item as EntityMenuItem);
+      }
+      return item as MenuItem;
```

Comment:
can you add a unit test that ensures "simple" menu items stay unchanged and "EntityMenuItem" gets parsed as expected?